docs/library/rp2.DMA.rst

```diff
@@ -0,0 +1,302 @@
+
+    src_data = bytearray(1024)
+    d = rp2.DMA()
+
+    # Transfer bytes, rather than words, don't increment the write address and pace the transfer.
+    c = d.pack_ctrl(size=0, inc_write=False, treq_sel=DATA_REQUEST_INDEX)
+
+    d.config(
+        read=src_data,
+        write=my_state_machine,
+        count=len(src_data),
+        ctrl=c,
+        trigger=True
+    )
+
+Note that in this example the value given for the write address is just the PIO state machine to
+which we are sending the data. This works because PIO state machines present the buffer protocol,
+allowing direct access to their data FIFO registers.
+
+
+Constructor
+-----------
+
+.. class:: DMA()
+
+    Claim one of the DMA controller channels for exclusive use.
+
+
+Methods
+-------
+
+.. method:: DMA.config(read=None, write=None, count=None, ctrl=None, trigger=False)
+
+    Configure the DMA registers for the channel and optionally start the transfer.
+
+    :param read: The address from which the DMA controller will start reading data or
+      an object that will provide data to be read. It can be an integer or any
+      object that supports the buffer protocol.
+    :param write: The address to which the DMA controller will start writing or an
+      object into which data will be written. It can be an integer or any object
+      that supports the buffer protocol.
+    :param count: The number of bus transfers that will execute before this channel
+      stops. Note that this is the number of transfers, not the number of bytes.
+      If the transfers are 2 or 4 bytes wide then the total amount of data moved
+      (and thus the size of required buffer) needs to be multiplied accordingly.
+    :param ctrl: The value for the DMA control register. This is an integer value
+      that is typically packed using the :meth:`DMA.pack_ctrl()`.
+    :param trigger: Optionally commence the transfer immediately.
+
+
+.. method:: DMA.irq(handler=None, hard=False)
+
+    Returns the IRQ object for this DMA channel and optionally configures it.
+
+.. method:: DMA.close()
+
+    Release the claim on the underlying DMA channel and free the interrupt
+    handler. The :class:`DMA` object can not be used after this operation.
+
+.. method:: DMA.pack_ctrl(default=None, **kwargs)
+
+    Pack the values provided in the keyword arguments into the named fields of a new control
+    register value. Any field that is not provided will be set to a default value. The
+    default will either be taken from the provided ``default`` value, or if that is not
+    given, a default suitable for the current channel; setting this to the current value
+    of the `DMA.ctrl` attribute provides an easy way to override a subset of the fields.
+
+    The keys for the keyword arguments can be any key returned by the :meth:`DMA.unpack_ctrl()`
+    method. The writable values are:
+
+    :param enable: ``BOOL`` Set to enable the channel (default: ``True``).
```

Comment:
Nitpick: Suggest `bool` or `boolean` for these, to be inline with other docs and/or the Python type.

Reply:
I have made this change during the rebase/merge.